File: modules/agent/src/mocks/controllers/reportController.ts
import { Router, Request, Response } from 'express';
import * as inventoryService from '../services/inventoryService.js';

export const reportRouter = Router();

reportRouter.get('/reports/low-stock', (_req: Request, res: Response) => {
  res.json(inventoryService.getLowStockAlerts());
});

reportRouter.get('/reports/dead-stock', (req: Request, res: Response) => {
  const days = parseInt(req.query.days as string, 10) || 90;
  res.json(inventoryService.getDeadStock(days));
});

File: modules/agent/src/agent/mocks/controllers/report_controller.py
from fastapi import APIRouter, Query

from agent.mocks.services import inventory_service

router = APIRouter(prefix="/api/reports", tags=["reports"])


@router.get("/low-stock")
async def get_low_stock_report():
    return inventory_service.get_low_stock_alerts()


@router.get("/dead-stock")
async def get_dead_stock_report(days: int = Query(90)):
    return inventory_service.get_dead_stock(days)
